docs/examples/aggregate.ipynb

```diff
@@ -41,29 +41,16 @@
     }
    ],
    "source": [
-    "from ipyvizzu import Chart, Data, Config\n",
-    "\n",
-    "data = Data()\n",
-    "\n",
-    "data.add_dimension('Genres', [ 'Pop', 'Rock', 'Jazz', 'Metal'])\n",
-    "data.add_dimension('Types', [ 'Hard', 'Smooth', 'Experimental' ])\n",
-    "\n",
-    "data.add_measure(\n",
-    "    'Popularity',\n",
-    "    [\n",
-    "        [114, 96, 78, 52],\n",
-    "        [56, 36, 174, 121],\n",
-    "        [127, 83, 94, 58],\n",
-    "    ]\n",
-    ")\n",
+    "from ipyvizzu import Chart, Config\n",
+    "from data_examples import music_example_data\n",
```

Comment:
How does know the reader of the online documentation what is the `music_example_data`?

Reply:
I use the 3rd example from data.ipynb

- But renamed it from data_cube to music_example_data,
because in the Data chapter, the format of the data is the most important,
but in the other chapters the data itself (in the future it could be more data for the examples
and all could be in data_cube format).

- I do not like the solutions, if examples does not work if the first cell is unplayed

What do you think about this:

- Create a Data Examples notebook, which contains all of the test data (in the future it could be more).
In the other notebooks I put a link to this notebook in the first cell's description.
This is also redundant, but data will be exist only two copies (in Examples notebook, and in data_examples.py) not in every notebook first cell?

Reply:
What if the test data would be data (json). The `Data` class could have `from_from` classmethod which initiates itself from a json file:

```python
class Data(Animation):
    @classmethod
    def from_json(cls, filename):
        with open(filename) as file_desc:
            return cls(json.load(file_desc))
```

music_example_data.json
```json
{
    "dimensions": [
        {"name": "Genres", "values": [ "Pop", "Rock", "Jazz", "Metal"]},
        {"name": "Types", "values": [ "Hard", "Smooth", "Experimental"]},
    ],
   "measure": [
       {
           "name": "Popularity",
           "values":  [
                [114, 96, 78, 52],
                [56, 36, 174, 121],
                [127, 83, 94, 58]
           ]
       }
   ]
}
```

Example:
```
from ipyvizzu import Data

data = Data.from_json("music_example_data.json")
```

The test doesn't have to modify the python path. Only the `cwd` has to be changed for the relative path of the json file.

Reply:
@nyirog I don't clearly understand, how change only the json file's cwd?

Reply:
My mistake. I thought that `jupyter notebook` update the current-working-directory to the directory of the `ipynb` file. But `jupyter notebook` does not make any such trickeries. So the `cwd` does not have to be changed because both the tests and the notebooks are run from the project root directory. The path of the json file has to be relative to the project root directory.

Reply:
My environment when I run a cell, the cdw is the path of the ipynb file, not the project root. Probably it is my problem, I'll do some research.

Reply:
Of course it would be better to validate the json data

```python
from jsonschema import validate

NAMED_VALUES_SCHEMA = {
    "type": "array",
    "items": {
        "type": "object",
        "properties": {
            "name": {
                "type": "string"
            },
            "values": {
                "type": "array"
            }
        },
        "required": ["name", "values"]
    } 
}

SCHEMA = {
    "type": "object",
    "properties": {
        "records": {
            "type": "array",
        },
        "series": NAMED_VALUES_SCHEMA,
        "measures": NAMED_VALUES_SCHEMA,
        "dimensions": NAMED_VALUES_SCHEMA,
    }
}

class Data(Animation):
    @classmethod
    def from_json(cls, filename):
        with open(filename) as file_desc:
            data = json.load(file_desc)

       validate(data, SCHEMA)
       return cls(data)
```

The validation does not have to be jsonschema. We could use `typing.NamedTuple`, `dataclasses` or `attrs`, but they unfortunately don't have `from_dict` method .

Reply:
Sorry, I made a mistake during testing. The `cwd` in `jupiter notebook` is the directory of the selected `ipynb` file like you said.